packages/ketcher-core/src/application/editor/operations/monomer/PresetAdd.ts

```diff
@@ -0,0 +1,182 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+/* eslint-disable @typescript-eslint/no-use-before-define */
+
+import { Vec2 } from 'domain/entities';
+import { ReStruct } from '../../../render';
+
+import { BaseOperation } from '../base';
+import { OperationType } from '../OperationType';
+import { MonomerItemType } from 'domain/types';
+
+import assert from 'assert';
+
+import { monomerFactory } from './monomerFactory';
+import { fromPolymerBondAddition } from 'application/editor/actions/polymerBond';
+import { BondService } from 'domain/services/bond/BondService';
+import { PolymerBondRenderer } from 'application/render/renderers/PolymerBondRenderer';
+import { BaseMonomer } from 'domain/entities/BaseMonomer';
+
+type Data = {
+  sugar: MonomerItemType;
+  sugarPosition: Vec2 | undefined;
+  rnaBase: MonomerItemType | undefined;
+  rnaBasePosition: Vec2 | undefined;
+  phosphate: MonomerItemType | undefined;
+  phosphatePosition: Vec2 | undefined;
+};
+
+class PresetAdd extends BaseOperation {
+  data: Data;
+  renderersContainer: ReStruct;
+  private bondRenderer?: PolymerBondRenderer; // only for sugar, in two sides
+
+  constructor(renderersContainer: ReStruct, presets: Data) {
+    super(OperationType.ATOM_ADD);
+    this.data = presets;
+    this.renderersContainer = renderersContainer;
+  }
+
+  execute(restruct: ReStruct) {
+    const {
+      sugar,
+      sugarPosition,
+      phosphate,
+      phosphatePosition,
+      rnaBase,
+      rnaBasePosition,
+    } = this.data;
+
+    const struct = restruct.molecule;
+
+    const [Sugar, SugarRenderer] = monomerFactory(sugar);
+    const newSugar = new Sugar(sugar, sugarPosition);
+    const sugarRenderer = new SugarRenderer(newSugar);
+    struct.monomers.set(newSugar.id, newSugar);
+    restruct.monomers.set(newSugar.id, sugarRenderer);
+
+    let rnaBaseRenderer;
+    let phosphateRenderer;
+    if (rnaBase && rnaBasePosition) {
+      const [RNABase, RNABaseRenderer] = monomerFactory(rnaBase);
+      const newRNABase = new RNABase(rnaBase, rnaBasePosition);
+      rnaBaseRenderer = new RNABaseRenderer(newRNABase);
+      struct.monomers.set(newRNABase.id, newRNABase);
+      restruct.monomers.set(newRNABase.id, rnaBaseRenderer);
+    }
+
+    if (phosphate && phosphatePosition) {
+      const [Phosphate, PhosphateRenderer] = monomerFactory(phosphate);
+      const newPhosphate = new Phosphate(phosphate, phosphatePosition);
+      phosphateRenderer = new PhosphateRenderer(newPhosphate);
+      struct.monomers.set(newPhosphate.id, newPhosphate);
+      restruct.monomers.set(newPhosphate.id, phosphateRenderer);
+    }
+
```

Comment:
I think we need somehow generalize code here to reduce duplication and to have an ability to create any amount of monomers in group. Let me try to do that when I will merge your changes in my pr https://github.com/epam/ketcher/pull/3217

Reply:
Yes^ cool, thank you!
Also I see now, that i do not need factory here, because I know for sure that it will be phosphate for example xd